cylc/flow/task_pool.py

```diff
@@ -1374,12 +1384,11 @@ def remove_if_complete(self, itask):
             if self.compute_runahead():
                 self.release_runahead_tasks()
         else:
-            incomplete = itask.state.outputs.get_incomplete()
-            if incomplete:
+            if itask.state.outputs.is_incomplete():
                 # Retain as incomplete.
                 LOG.warning(
                     f"[{itask}] did not complete required outputs:"
-                    f" {incomplete}"
+                    f" {itask.state.outputs.get_incomplete()}"
```

Comment:
After these changes `bool(get_incomplete()) != is_complete()`.

E.G. if `completion = succeeded and (x or y)` and `succeeded` and `x` have been generated, `get_incomplete()` will return `y`.